.github/workflows/ci.yml

```diff
@@ -47,13 +47,14 @@ jobs:
         with:
           python-version-file: "pyproject.toml"
           cache: "poetry"
-
+      - name: Install poetry with pip
+        run: python -m pip install poetry
```

Comment:
this seems unrelated to the rest of the pr. Is it an intentional change?

Reply:
this could be unnecessary but it was introduced to fix a strange issue adding pypdfium2 uncovered with the python version our ci runner uses (it had been using 3.10 all along and not the pinned 3.12). will see if i can take this step out